FWCore/Framework/src/Path.h

```diff
@@ -109,19 +108,24 @@ namespace edm {
     int timesFailed_;
     int timesExcept_;
     //int abortWorker_;
-    State state_;
-
-    int bitpos_;
-    TrigResPtr trptr_;
-    std::shared_ptr<ActivityRegistry> actReg_;  // We do not use propagate_const because the registry itself is mutable.
-    ExceptionToActionTable const* act_table_;
+    //When an exception happens, it is possible for multiple modules in a path to fail
+    // and then try to change the state concurrently.
+    std::atomic<bool> stateLock_ = false;
+    CMS_THREAD_GUARD(stateLock_) int failedModuleIndex_;
+    CMS_THREAD_GUARD(stateLock_) State state_;
+
+    int const bitpos_;
+    TrigResPtr const trptr_;
+    std::shared_ptr<ActivityRegistry> const
+        actReg_;  // We do not use propagate_const because the registry itself is mutable.
```

Comment:
```suggestion
    // We do not use propagate_const because the registry itself is mutable.
    std::shared_ptr<ActivityRegistry> const actReg_;
```